File: app/src/main/java/com/roy/downloader/receiver/BootReceiver.java
package com.roy.downloader.receiver;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

import com.roy.downloader.core.RepositoryHelper;
import com.roy.downloader.core.model.DownloadEngine;
import com.roy.downloader.core.settings.SettingsRepository;

/*
 * The receiver for autostart stopped downloads.
 */

public class BootReceiver extends BroadcastReceiver {
    @Override
    public void onReceive(Context context, Intent intent) {
        if (intent.getAction() == null)
            return;

        if (intent.getAction().equals(Intent.ACTION_BOOT_COMPLETED)) {
            SettingsRepository pref = RepositoryHelper.getSettingsRepository(context.getApplicationContext());
            if (pref != null && pref.autostart()) {
                DownloadEngine engine = DownloadEngine.getInstance(context.getApplicationContext());
                engine.restoreDownloads();
            }
        }
    }
}

File: app/src/main/java/com/roy/downloader/receiver/BootReceiver.kt
package com.roy.downloader.receiver

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import com.roy.downloader.core.RepositoryHelper.getSettingsRepository
import com.roy.downloader.core.model.DownloadEngine

/*
 * The receiver for autostart stopped downloads.
 */
class BootReceiver : BroadcastReceiver() {
    override fun onReceive(context: Context, intent: Intent) {
        if (intent.action == null) return
        if (intent.action == Intent.ACTION_BOOT_COMPLETED) {
            val pref = getSettingsRepository(context.applicationContext)
            if (pref != null && pref.autostart()) {
                val engine = DownloadEngine.getInstance(context.applicationContext)
                engine.restoreDownloads()
            }
        }
    }
}
